app/src/main/kotlin/com/wire/android/util/permission/CurrentLocationRequestFlow.kt

```diff
@@ -29,34 +29,47 @@ import com.wire.android.util.extension.checkPermission
 
 @Composable
 fun rememberCurrentLocationFlow(
-    onLocationPicked: () -> Unit, // TODO: this will change accordingly to maps intent
+    onPermissionAllowed: () -> Unit,
     onPermissionDenied: () -> Unit
 ): CurrentLocationRequestFlow {
     val context = LocalContext.current
 
-    val requestPermissionLauncher: ManagedActivityResultLauncher<String, Boolean> =
-        rememberLauncherForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted ->
-            if (isGranted) {
-                // TODO: launch map location picker using openstreetmap? (have in mind f-droid aka. no gms)
+    val requestPermissionLauncher: ManagedActivityResultLauncher<Array<String>, Map<String, @JvmSuppressWildcards Boolean>> =
+        rememberLauncherForActivityResult(ActivityResultContracts.RequestMultiplePermissions()) { permissions ->
+            val allPermissionGranted = permissions.all { it.value }
+            if (allPermissionGranted) {
+                onPermissionAllowed()
             } else {
                 onPermissionDenied()
             }
```

Comment:
Thanks, good point. 
According to [designs](https://www.figma.com/file/UgKehvwIbiIiQ2CCEdsVuf/Android?node-id=40141%3A75324&mode=dev), we need to show this dialog if the user declined previously (basically we are suggesting him to go to "app settings" with redirection). Also, this is the same behavior for other attachments like Recording Audio.

Reply:
Good idea about https://wearezeta.atlassian.net/browse/WPB-6154
I extracted a component, so we can reuse it at.

app/src/main/kotlin/com/wire/android/util/permission/PermissionsDeniedRequestDialog.kt

Reply:
I think it's not up to date
https://www.figma.com/file/UgKehvwIbiIiQ2CCEdsVuf/Android?type=design&node-id=10548-105517&mode=dev
Also it's recommended to do it this way
https://developer.android.com/training/permissions/requesting#handle-denial

Reply:
I'll leave it as is, since it's up-to-date for the sharing location MVP and congruent with the recording audio approach of handling (the most recent attachment implementation we have). Anyway, we will take a look in the design review for more comments from the design team in case changes are needed =)

Also, since a new ticket was created. I think the recommended way to handle the permissions and the figma file linked there is in the scope of this other ticket https://wearezeta.atlassian.net/browse/WPB-6154